The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8870196920/job/24351716951) [(plain)](https://github.com/rust-lang/rust/commit/ed9d6e0c03b53aea602ea9b0a283b39491f73b91/checks/24351716951/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:0ad4b8fadaa221de15dcec353f45205ec38ea70b)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.408   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.423 Collecting boolean-py==4.0
#13 0.427   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.440 Collecting chardet==5.1.0
---
#13 3.466 Building wheels for collected packages: reuse
#13 3.467   Building wheel for reuse (pyproject.toml): started
#13 3.784   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.785   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.785   Stored in directory: /tmp/pip-ephem-wheel-cache-as9pmzf2/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.788 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.809   Attempting uninstall: setuptools
#13 3.810     Found existing installation: setuptools 59.6.0
#13 3.811     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.811     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 3.811     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.487 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.487 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.984 Collecting virtualenv
#13 5.035   Downloading virtualenv-20.26.0-py3-none-any.whl (3.9 MB)
#13 5.080      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 91.6 MB/s eta 0:00:00
#13 5.117 Collecting distlib<1,>=0.3.7
#13 5.123   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.159 Collecting filelock<4,>=3.12.2
#13 5.164   Downloading filelock-3.13.4-py3-none-any.whl (11 kB)
#13 5.164   Downloading filelock-3.13.4-py3-none-any.whl (11 kB)
#13 5.192 Collecting platformdirs<5,>=3.9.1
#13 5.196   Downloading platformdirs-4.2.1-py3-none-any.whl (17 kB)
#13 5.283 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.445 Successfully installed distlib-0.3.8 filelock-3.13.4 platformdirs-4.2.1 virtualenv-20.26.0
#13 DONE 5.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      217024 kB
DirectMap2M:     7122944 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/ce609db948f210807f61da999c57bd4ae5216254/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-ce609db948f210807f61da999c57bd4ae5216254-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 25.45s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/std/tests/minmax-stability-issue-23687.rs at line 1:
 use std::cmp::{self, Ordering};
+use std::fmt::Debug;
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 struct Foo {
##[error]Diff in /checkout/library/std/tests/minmax-stability-issue-23687.rs at line 6:
     n: u8,
+    name: &'static str,
 }
 
 impl PartialOrd for Foo {
 impl PartialOrd for Foo {
##[error]Diff in /checkout/library/std/tests/istr.rs at line 9:
 
 #[test]
 #[test]
-fn test_heap_lit() { "a big string".to_string(); }
+fn test_heap_lit() {
+    "a big string".to_string();
 
 #[test]
 fn test_heap_assign() {
 fn test_heap_assign() {
##[error]Diff in /checkout/library/std/tests/log-knows-the-names-of-variants-in-std.rs at line 3:
 #[derive(Clone, Debug)]
 enum foo {
-  a(usize),
-  b(String),
-  b(String),
+    a(usize),
+    b(String),
 }
 
 fn check_log<T: std::fmt::Debug>(exp: String, v: T) {
##[error]Diff in /checkout/library/std/tests/eq-multidispatch.rs at line 7:
 struct Fu;
 
 
-impl PartialEq for Baz { fn eq(&self, _: &Baz) -> bool  { true } }
+impl PartialEq for Baz {
+    fn eq(&self, _: &Baz) -> bool {
+    }
+}
 
 
-impl PartialEq<Fu> for Foo { fn eq(&self, _: &Fu) -> bool { true } }
-impl PartialEq<Foo> for Fu { fn eq(&self, _: &Foo) -> bool { true } }
+impl PartialEq<Fu> for Foo {
+    fn eq(&self, _: &Fu) -> bool {
+    }
+}
+}
+impl PartialEq<Foo> for Fu {
+    fn eq(&self, _: &Foo) -> bool {
+    }
+}
 
 
-impl PartialEq<Bar> for Foo { fn eq(&self, _: &Bar) -> bool { false } }
-impl PartialEq<Foo> for Bar { fn eq(&self, _: &Foo) -> bool { false } }
+impl PartialEq<Bar> for Foo {
+    fn eq(&self, _: &Bar) -> bool {
+    }
+}
+}
+impl PartialEq<Foo> for Bar {
+    fn eq(&self, _: &Foo) -> bool {
+    }
+}
 
 #[test]
 #[test]
 fn eq_multidispatch() {
##[error]Diff in /checkout/library/std/tests/builtin-clone.rs at line 6:
     let _ = arg.clone();
 
-fn foo() { }
+fn foo() {}
 
 
 #[derive(Debug, PartialEq, Eq)]
 struct S(i32);
##[error]Diff in /checkout/library/std/tests/builtin-clone.rs at line 27:
     let b = [S(1), S(2), S(3)];
     assert_eq!(b, a.clone());
-    let a = (
-    let a = (
-        (S(1), S(0)),
-        (
-            (S(0), S(0), S(1)),
-            S(0)
-    );
-    let b = (
-    let b = (
-        (S(2), S(1)),
-        (
-            (S(1), S(1), S(2)),
-            S(1)
-    );
-    );
+    let a = ((S(1), S(0)), ((S(0), S(0), S(1)), S(0)));
+    let b = ((S(2), S(1)), ((S(1), S(1), S(2)), S(1)));
     assert_eq!(b, a.clone());
 
 
##[error]Diff in /checkout/library/std/tests/issue-21058.rs at line 3:
 
 struct NT(str);
 struct NT(str);
-struct DST { a: u32, b: str }
+struct DST {
+    a: u32,
+    b: str,
 
 macro_rules! check {
 macro_rules! check {
     (val: $ty_of:expr, $expected:expr) => {
##[error]Diff in /checkout/library/std/tests/issue-21058.rs at line 56:
 
 impl Foo {
 impl Foo {
-    fn new() -> Self { Foo }
+    fn new() -> Self {
+    }
 }
 
 fn foo() -> impl Debug {
 fn foo() -> impl Debug {
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/tests/minmax-stability-issue-23687.rs" "/checkout/library/std/tests/env.rs" "/checkout/library/std/tests/common/mod.rs" "/checkout/library/std/tests/create_dir_all_bare.rs" "/checkout/library/std/tests/volatile-fat-ptr.rs" "/checkout/library/std/tests/istr.rs" "/checkout/library/std/tests/switch-stdout.rs" "/checkout/library/std/tests/thread.rs" "/checkout/library/std/tests/run-time-detect.rs" "/checkout/library/std/tests/log-knows-the-names-of-variants-in-std.rs" "/checkout/library/std/tests/slice-from-array-issue-113238.rs" "/checkout/library/std/tests/process_spawning.rs" "/checkout/library/std/tests/seq-compare.rs" "/checkout/library/std/tests/eq-multidispatch.rs" "/checkout/library/std/tests/builtin-clone.rs" "/checkout/library/std/tests/issue-21058.rs" "/checkout/src/tools/jsondoclint/src/validator/tests.rs" "/checkout/src/tools/jsondoclint/src/main.rs" "/checkout/src/tools/jsondoclint/src/item_kind.rs" "/checkout/src/tools/jsondoclint/src/json_find.rs" "/checkout/src/tools/jsondoclint/src/json_find/tests.rs" "/checkout/src/tools/jsondoclint/src/validator.rs" "/checkout/src/tools/lld-wrapper/src/main.rs" "/checkout/src/tools/opt-dist/src/exec.rs" "/checkout/src/tools/opt-dist/src/timer.rs" "/checkout/src/tools/opt-dist/src/training.rs" "/checkout/src/tools/opt-dist/src/utils/artifact_size.rs" "/checkout/src/tools/opt-dist/src/utils/io.rs" "/checkout/src/tools/opt-dist/src/utils/mod.rs" "/checkout/src/tools/opt-dist/src/main.rs" "/checkout/src/tools/opt-dist/src/bolt.rs" "/checkout/src/tools/opt-dist/src/metrics.rs" "/checkout/src/tools/opt-dist/src/environment.rs" "/checkout/src/tools/opt-dist/src/tests.rs" "/checkout/src/tools/miropt-test-tools/src/lib.rs" "/checkout/library/std/src/sys/personality/emcc.rs" "/checkout/src/tools/jsondocck/src/error.rs" "/checkout/src/tools/jsondocck/src/main.rs" "/checkout/src/tools/jsondocck/src/cache.rs" "/checkout/src/tools/jsondocck/src/config.rs" "/checkout/library/std/src/sys/personality/dwarf/mod.rs" "/checkout/library/std/src/sys/personality/dwarf/eh.rs" "/checkout/library/std/src/sys/personality/dwarf/tests.rs" "/checkout/library/std/src/sys/personality/mod.rs" "/checkout/library/std/src/sys/personality/gcc.rs" "/checkout/library/std/src/sys/os_str/bytes.rs" "/checkout/library/std/src/sys/os_str/mod.rs" "/checkout/library/std/src/sys/os_str/bytes/tests.rs" "/checkout/library/std/src/sys/os_str/wtf8.rs" "/checkout/src/tools/build_helper/src/ci.rs" "/checkout/src/tools/build_helper/src/lib.rs" "/checkout/src/tools/build_helper/src/git.rs" "/checkout/src/tools/build_helper/src/util.rs" "/checkout/src/tools/build_helper/src/metrics.rs" "/checkout/src/tools/bump-stage0/src/main.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sun Apr 28 20:30:40 UTC 2024
  network time: Sun, 28 Apr 2024 20:30:40 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/tests/builtin-clone.rs
+++ b/library/std/tests/builtin-clone.rs
@@ -6,7 +6,7 @@ fn test_clone<T: Clone>(arg: T) {
     let _ = arg.clone();
 }
 
-fn foo() { }
+fn foo() {}
 
 #[derive(Debug, PartialEq, Eq)]
 struct S(i32);
@@ -27,19 +27,7 @@ fn builtin_clone() {
     let b = [S(1), S(2), S(3)];
     assert_eq!(b, a.clone());
 
-    let a = (
-        (S(1), S(0)),
-        (
-            (S(0), S(0), S(1)),
-            S(0)
-        )
-    );
-    let b = (
-        (S(2), S(1)),
-        (
-            (S(1), S(1), S(2)),
-            S(1)
-        )
-    );
+    let a = ((S(1), S(0)), ((S(0), S(0), S(1)), S(0)));
+    let b = ((S(2), S(1)), ((S(1), S(1), S(2)), S(1)));
     assert_eq!(b, a.clone());
 }

--- a/library/std/tests/eq-multidispatch.rs
+++ b/library/std/tests/eq-multidispatch.rs
@@ -7,13 +7,35 @@ struct Foo;
 #[derive(Debug)]
 struct Fu;
 
-impl PartialEq for Baz { fn eq(&self, _: &Baz) -> bool  { true } }
+impl PartialEq for Baz {
+    fn eq(&self, _: &Baz) -> bool {
+        true
+    }
+}
 
-impl PartialEq<Fu> for Foo { fn eq(&self, _: &Fu) -> bool { true } }
-impl PartialEq<Foo> for Fu { fn eq(&self, _: &Foo) -> bool { true } }
+impl PartialEq<Fu> for Foo {
+    fn eq(&self, _: &Fu) -> bool {
+        true
+    }
+}
 
-impl PartialEq<Bar> for Foo { fn eq(&self, _: &Bar) -> bool { false } }
-impl PartialEq<Foo> for Bar { fn eq(&self, _: &Foo) -> bool { false } }
+impl PartialEq<Foo> for Fu {
+    fn eq(&self, _: &Foo) -> bool {
+        true
+    }
+}
+
+impl PartialEq<Bar> for Foo {
+    fn eq(&self, _: &Bar) -> bool {
+        false
+    }
+}
+
+impl PartialEq<Foo> for Bar {
+    fn eq(&self, _: &Foo) -> bool {
+        false
+    }
+}
 
 #[test]
 fn eq_multidispatch() {

--- a/library/std/tests/issue-21058.rs
+++ b/library/std/tests/issue-21058.rs
@@ -3,7 +3,11 @@
 use std::fmt::Debug;
 
 struct NT(str);
-struct DST { a: u32, b: str }
+
+struct DST {
+    a: u32,
+    b: str,
+}
 
 macro_rules! check {
     (val: $ty_of:expr, $expected:expr) => {
@@ -56,7 +60,9 @@ fn type_name_of_val<T>(_: T) -> &'static str {
 struct Foo;
 
 impl Foo {
-    fn new() -> Self { Foo }
+    fn new() -> Self {
+        Foo
+    }
 }
 
 fn foo() -> impl Debug {

--- a/library/std/tests/istr.rs
+++ b/library/std/tests/istr.rs
@@ -9,7 +9,9 @@ fn test_stack_assign() {
 }
 
 #[test]
-fn test_heap_lit() { "a big string".to_string(); }
+fn test_heap_lit() {
+    "a big string".to_string();
+}
 
 #[test]
 fn test_heap_assign() {

--- a/library/std/tests/log-knows-the-names-of-variants-in-std.rs
+++ b/library/std/tests/log-knows-the-names-of-variants-in-std.rs
@@ -3,8 +3,8 @@
 
 #[derive(Clone, Debug)]
 enum foo {
-  a(usize),
-  b(String),
+    a(usize),
+    b(String),
 }
 
 fn check_log<T: std::fmt::Debug>(exp: String, v: T) {

--- a/library/std/tests/minmax-stability-issue-23687.rs
+++ b/library/std/tests/minmax-stability-issue-23687.rs
@@ -1,10 +1,10 @@
-use std::fmt::Debug;
 use std::cmp::{self, Ordering};
+use std::fmt::Debug;
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq)]
 struct Foo {
     n: u8,
-    name: &'static str
+    name: &'static str,
 }
 
 impl PartialOrd for Foo {

--- a/library/std/tests/type-name-unsized.rs
+++ b/library/std/tests/type-name-unsized.rs
@@ -18,24 +18,24 @@ macro_rules! check {
     };
 }
 
+/// Tests that [`std::any::type_name`] supports unsized types.
 #[test]
-fn issue_21058() {
-    // type_name should support unsized types
+fn type_name_unsized() {
     check!([u8], "[u8]");
     check!(str, "str");
     check!(dyn Send, "dyn core::marker::Send");
-    check!(NT, "issue_21058::NT");
-    check!(DST, "issue_21058::DST");
+    check!(NT, "type_name_unsized::NT");
+    check!(DST, "type_name_unsized::DST");
     check!(&i32, "&i32");
     check!(&'static i32, "&i32");
     check!((i32, u32), "(i32, u32)");
-    check!(val: foo(), "issue_21058::Foo");
-    check!(val: Foo::new, "issue_21058::Foo::new");
+    check!(val: foo(), "type_name_unsized::Foo");
+    check!(val: Foo::new, "type_name_unsized::Foo::new");
     check!(val:
         <Foo as Debug>::fmt,
-        "<issue_21058::Foo as core::fmt::Debug>::fmt"
+        "<type_name_unsized::Foo as core::fmt::Debug>::fmt"
     );
-    check!(val: || {}, "issue_21058::issue_21058::{{closure}}");
+    check!(val: || {}, "type_name_unsized::type_name_unsized::{{closure}}");
     bar::<i32>();
 }
 

--- a/src/tools/tidy/src/issues.txt
+++ b/src/tools/tidy/src/issues.txt
@@ -3862,7 +3862,6 @@ ui/statics/issue-91050-1.rs
 ui/statics/issue-91050-2.rs
 ui/std/issue-3563-3.rs
 ui/std/issue-81357-unsound-file-methods.rs
-ui/stdlib-unit-tests/issue-21058.rs
 ui/structs-enums/enum-rec/issue-17431-6.rs
 ui/structs-enums/enum-rec/issue-17431-7.rs
 ui/structs-enums/issue-103869.rs

